Runtime/SharedResources/Scripts/Vrtk4UiToPointer.cs

```diff
@@ -1,139 +1,173 @@
+            ChangePointsCastEventDataValidState(true);
+
+            pointerFacade.Configuration.EmitSelected(PrepareSelectEventDataObject(HoverEnterDuration, true, true));
         }
-        
-        private void OnHoverExit(VRTK4_UIPointer interactable, VRTK4_UIPointer.VRTK4UIPointerEventArgs eventdata)
+
+        private void OnSelectExit(VRTK4_UIPointer.VRTK4UIPointerEventArgs eventData)
         {
-            OnSelectExit(interactable, eventdata);
-            
-            temporalList.Clear();
-            temporalList.Add(eventdata.raycastResult.worldPosition);
-            
+            CacheRaycastResultWorldPosition(eventData.raycastResult.worldPosition);
             FillInEventData();
-            pointerCastEventData.IsValid = false;
-            var data = eventData;
-            data.CurrentHoverDuration = 0.9f;
-            data.IsCurrentlyHovering = false;
-            data.CurrentPointsCastData = pointerCastEventData;
-            data.CurrentPointsCastData.Points = new HeapAllocationFreeReadOnlyList<Vector3>(temporalList,0,temporalList.Count);
-            pointerFacade.Configuration.EmitHoverChanged(data);
+            ChangePointsCastEventDataValidState(true);
+
+            pointerFacade.Configuration.EmitExited(PrepareSelectEventDataObject(HoverExitDuration, false, false));
+        }
+
+        private bool ValidatePointerData(VRTK4_UIPointer interactable)
+        {
+            return interactable != null && pointerFacade != null;
+        }
+
+        private void CacheRaycastResultWorldPosition(Vector3 worldPosition)
+        {
+            cachedRaycastResultWorldPositions.Clear();
+            cachedRaycastResultWorldPositions.Add(worldPosition);
         }
         
-        private void OnSelectEnter(VRTK4_UIPointer interactable, VRTK4_UIPointer.VRTK4UIPointerEventArgs eventdata)
+        /// <summary>
+        /// Generates all of the data for pointerFacade
+        /// </summary>
+        private void FillInEventData()
         {
-            if (interactable == null)
+            pointsCastEventData ??= new PointsCast.EventData();
+
+            objectPointerEventData ??= new ObjectPointer.EventData
             {
-                return;
-            }
-            temporalList.Clear();
-            temporalList.Add(eventdata.raycastResult.worldPosition);
-            FillInEventData();
-            pointerCastEventData.IsValid = true;
-            var data = eventData;
-            data.CurrentHoverDuration = 0.1f;
-            data.IsCurrentlyHovering = true;
-            data.IsCurrentlyActive = true;
-            data.CurrentPointsCastData = pointerCastEventData;
-            data.CurrentPointsCastData.Points = new HeapAllocationFreeReadOnlyList<Vector3>(temporalList,0,temporalList.Count);
-            pointerFacade.Configuration.EmitSelected(data);
+                Transform = transform
+            };
+
+            objectPointerEventData.Clear();
```

Comment:
@ArtemkaKun  also, need to clear pointsCastEventData

Reply:
Ok